### PR TITLE
qa/tasks: set defer_client_eviction_on_laggy_osds=false in api tests

### PR DESCRIPTION
**Pull request by ljflores**
We expect laggy OSDs in this testing environment,
so it makes sense to whitelist this warning.

Fixes: https://tracker.ceph.com/issues/61907

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by batrick** [comment]
https://tracker.ceph.com/issues/61907#note-9

**Comment by ljflores** [comment]
Whitelisting where I tried didn't work, so trying something else.

**Comment by dparmar18** [comment]
@ljflores any update on this? do ping me in case you need any help

**Comment by ljflores** [comment]
Couldn't find a fancy way to whitelist, but this works too. If anyone knows of a better solution, I'm willing to modify it.

**Comment by dparmar18** [comment]
> Couldn't find a fancy way to whitelist, but this works too. If anyone knows of a better solution, I'm willing to modify it.


https://github.com/ceph/ceph/blob/main/qa/tasks/mgr/dashboard/helper.py#L271

An API test will wait for 60 secs for the cluster to report healthy; and this laggy OSD warning will not off that easily; seems like we cannot override it with some yaml. I think a better solution here is to turn off the config option `defer_client_eviction_on_laggy_osds` in testing env

File: qa/tasks/ceph_test_case.py [hunk 1]
```diff
@@ -38,6 +38,7 @@ class CephTestCase(unittest.TestCase):
 
     def setUp(self):
         self._mon_configs_set = set()
+        self.config_set('mds', 'defer_client_eviction_on_laggy_osds', 'false')
```
Comment (nizamial09) [inline qa/tasks/ceph_test_case.py:41]: @dparmar18 @batrick while this does fix/suppress the api test failure, i am more curious on how this newly added health check `MDS_CLIENTS_LAGGY` will clear itself. Is this error always going to be present in the test environment? And also how's an OSD considered laggy?

**Comment by ljflores** [comment]
api test passed. Retriggering one more time for good measure.

**Comment by ljflores** [comment]
jenkins test api

File: qa/tasks/ceph_test_case.py (hunk 1 above)
Comment (ljflores) [reply qa/tasks/ceph_test_case.py:41]: @batrick added some explanation here: https://tracker.ceph.com/issues/61907#note-9

Comment (batrick) [reply qa/tasks/ceph_test_case.py:41]: Let's move this to `qa/tasks/mgr/mgr_test_case.py` (sp?) instead.

**Review by batrick — CHANGES_REQUESTED** [review]

File: qa/tasks/mgr/mgr_test_case.py [hunk 2]
```diff
@@ -105,6 +105,8 @@ def setUpClass(cls):
                 "Only have {0} manager daemons, {1} are required".format(
                     len(cls.mgr_cluster.mgr_ids), cls.MGRS_REQUIRED))
 
+        cls.mgr_cluster.mon_manager.raw_cluster_cmd('config', 'set', 'mds',
+                                                    'defer_client_eviction_on_laggy_osds', 'false')
```
Comment (dparmar18) [inline qa/tasks/mgr/mgr_test_case.py:109]: having a comment explaining the reason to turn it off would be good

Comment (ljflores) [reply qa/tasks/mgr/mgr_test_case.py:109]: Sure

**Review by batrick — APPROVED** [review]

**Review by dparmar18 — APPROVED** [review]

**Review by nizamial09 — APPROVED** [review]

**Comment by nizamial09** [comment]
merging this now since it'll unblock the test failures. and thank you @ljflores for the fix!

**Comment by nizamial09** [comment]
@dparmar18 please update the tracker if you think this needs to go in to other branches like reef or quincy.

**Comment by sseshasa** [comment]
@ljflores This needs to be backported along with #52270. We can include both in a separate QA batch.

**Pull request merged**